zson/formatter.go

```diff
@@ -341,19 +360,22 @@ func (f *Formatter) formatMap(indent int, typ *zed.TypeMap, bytes zcode.Bytes, k
 	f.build("|{")
 	indent += f.tab
 	sep := f.newline
+	keyElems := newElemBuilder(typ.KeyType)
+	valElems := newElemBuilder(typ.ValType)
 	for it := bytes.Iter(); !it.Done(); {
-		keyBytes := it.Next()
+		kbytes := it.Next()
 		if it.Done() {
 			return empty, errors.New("truncated map value")
```

Comment:
This check was already there but I'm think this should be a panic, no?

Reply:
I'd just leave it.